### ISSUE TITLE
Parameter recommendations

### ISSUE THREAD

**Issue by wastacat** [pierluigiferrari/ssd_keras#73]
@pierluigiferrari : Hello pierluigiferrari   I am currently using ssd7 training face dataset， I want to do a real-time multi-face detection , video streaming size 600*800 , but I encountered a difficult point , As the surveillance camera hung on the wall, cause the face target size smaller.

Can you give me some advice about Training related parameters ,like scales , aspect_ratios , variances how to setting or more suggestion ?

Thanks in advance : )

**Comment by pierluigiferrari** [comment]
Just a few quick points:
1. Since you're trying to detect faces only, which are roughly square, you could try going with aspect ratio 1:1 as the only aspect ratio.
2. When it comes to the sizes and the distribution of your anchor boxes across the images, there are two you want to make sure:
    1. The anchor boxes should cover the scales of the ground truth boxes, so make the scaling factors however small you need to in order to cover the ground truth.
    2. The smaller your objects (and thus your anchor boxes), the more anchor boxes you will want spatially in order to cover the whole image appropriately. How do you get more anchor boxes in the spatial dimensions? You increase the spatial dimensions of the convolutional predictor layers. I order to do that, you could experiment with making predictions from an earlier conv layer, e.g. conv3. Or you could remove one of the pooling layers, but then of course all predictor layers that come after it will quadruple in their spatial sizes.
Especially for this second point, you cannot get around really putting in the effort and calculating and maybe visualizing what your grid of anchor boxes looks like for a given predictor layer. And then you can answer the questions: Is this a good coverage of my input image? Do the sizes and aspect ratios of these anchor boxes make sense for the training data I have? Do these anchor boxes even lie within the receptive fields of their respective layers?
3. Over the next weeks I'll implement [Focal Loss](https://arxiv.org/abs/1708.02002), which might improve SSD's results for small objects. I don't have the time to do it right now, but I will implement and try it out eventually.
4. When it comes to the input images size, you have the following problem: The smaller the objects are that you want to detect, the more important it becomes to use higher resolution images. This should be obvious: If you want to retain a certain level of detail of very small objects, the image resolution has to be appropriate. At the same time though, increasing image size makes your network drastically slower. I don't know what the answer is in your case. On the one hand, using 600*800 input (or as large as you can get, really) is great for the prediction quality, but at the same time you have to see up to what resolution your network can still make predictions at the frame rate you need, whatever that is.

**Comment by wastacat** [comment]
@pierluigiferrari  Thank you for your reply , I will try your advice.
